client/src/config.rs

```diff
@@ -0,0 +1,97 @@
+//! Module for client-related configuration and structs
+
+// FIXME
+#![allow(unused, missing_docs)]
+
+use core::str::FromStr;
+use std::{num::NonZeroU64, path::Path, time::Duration};
+
+use derive_more::Display;
+use eyre::Result;
+pub use iroha_config::base;
+use iroha_config::base::UnwrapPartial;
+use iroha_crypto::prelude::*;
+use iroha_data_model::{prelude::*, ChainId};
+use iroha_primitives::small::SmallStr;
+use serde::{Deserialize, Serialize};
+use url::Url;
+
+use crate::config::user_layer::RootPartial;
+
+pub mod user_layer;
```

Comment:
there is lot's of structs inside this module that I don't want exposed to the end user. Take that into consideration

Reply:
Hm, why not?

Reply:
@mversic it's not entirely what you don't what to expose, could you please specify?

Reply:
> Hm, why not?

because they look like implementation details to me. Will user ever need them?

Reply:
> @mversic it's not entirely what you don't what to expose, could you please specify?

for example, `pub struct Root`. This looks like it should be encapsulated and not part of the public API

Reply:
Hm, I had an idea that user-layer structures might be useful if the client's user wants to customise config loading procedure in some way. Like, read multiple files and merge them...

Well, now it makes more sense to hide it due to absence of clear use case.